Hackernews search in EVADB

kaushikravichandran opened the pull request:
This PR supports searching Hackernews stories in EVADB:

- We treat Hackernews as a data source
- Algolia search API for hackernews is used for searching
- Get requests are used
- We support searching within: `story`, `comment`, `URL`, `story`, `poll` or a combination of them
- Syntax:
```
params = {
        "query": "EVADB",
        "tags": "(story,poll)",
    }
query = f"""CREATE DATABASE hackernews_data
    WITH ENGINE = "hackernews",
    PARAMETERS = {params};"""
```

xzdandy commented:
Thanks for adding Hackernews as data source? Could you also add a documentation page for it? 
https://evadb.readthedocs.io/en/stable/source/reference/databases/index.html

kaushikravichandran commented:
> Thanks for adding Hackernews as data source? Could you also add a documentation page for it? https://evadb.readthedocs.io/en/stable/source/reference/databases/index.html

Added the docs. The docs test fail now: because I have added a link to the column info of the hackernews search results table: which is in this PR, and the link will become active once the PR is merged.
Link: https://github.com/georgia-tech-db/evadb/blob/staging/evadb/third_party/databases/hackernews/table_column_info.py

xzdandy commented on docs/source/reference/databases/hackernews.rst:
```diff
@@ -0,0 +1,44 @@
+
+   CREATE DATABASE hackernews_data WITH ENGINE = 'hackernews', PARAMETERS = {
+        "owner": "EVADB",
+        "repo": "story"
```
I think it should be query and tag instead of owner and repo.

kaushikravichandran replied:
Thanks! fixed it.

xzdandy reviewed (approved):
Looks great! Thanks.
